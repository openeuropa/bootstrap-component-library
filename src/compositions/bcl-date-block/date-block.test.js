import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-date-block/data/data";
import demoEndDateData from "@openeuropa/bcl-date-block/data/data--end-date";

const template = "@oe-bcl/bcl-date-block/date-block.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Date block", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("with end date renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoEndDateData)).resolves.toMatchSnapshot();
  });
});
