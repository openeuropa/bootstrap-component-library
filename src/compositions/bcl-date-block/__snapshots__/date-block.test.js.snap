// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Date block renders correctly 1`] = `
<jest>
  <time
    class="bcl-date-block d-flex flex-column align-items-center justify-content-center bg-date text-light pt-2 rounded-top mw-date"
    datetime="2021-01-23"
  >
    <span
      class="py-4 bg-white border border-bottom-0 w-100 fs-5 fw-medium text-dark text-center text-uppercase"
    >
      23 Jan
    </span>
    <span
      class="bg-lighter border w-100 text-center fs-5 text-dark rounded-bottom mb-n1"
    >
      2021
    </span>
  </time>
</jest>
`;
